litex/soc/interconnect/avalon.py

```diff
@@ -9,6 +9,156 @@
 from migen import *
 
 from litex.soc.interconnect import stream
+from . import wishbone
+
+_layout = [
+    ("address",          "adr_width", DIR_M_TO_S),
+    ("writedata",       "data_width", DIR_M_TO_S),
+    ("readdata",        "data_width", DIR_S_TO_M),
+    ("readdatavalid",              1, DIR_S_TO_M),
+    ("byteenable",       "sel_width", DIR_M_TO_S),
+    ("read",                       1, DIR_M_TO_S),
+    ("write",                      1, DIR_M_TO_S),
+    ("waitrequest",                1, DIR_S_TO_M),
+    ("burstbegin",                 1, DIR_M_TO_S), # this is optional
+    ("burstcount",                 8, DIR_M_TO_S),
+    ("chipselect",                 1, DIR_M_TO_S), # this is optional
+]
+
+class MMInterface(Record):
```

Comment:
OK, I actually did it this way to make it consistent with wishbone.Interface,
so it would be avalon.MMInterface, but I don't have a problem with this naming also.

Reply:
eon